crates/polars-core/src/datatypes/dtype.rs

```diff
@@ -542,11 +543,15 @@ impl DataType {
             Float32 => Ok(ArrowDataType::Float32),
             Float64 => Ok(ArrowDataType::Float64),
             #[cfg(feature = "dtype-decimal")]
-            // note: what else can we do here other than setting precision to 38?..
-            Decimal(precision, scale) => Ok(ArrowDataType::Decimal(
-                (*precision).unwrap_or(38),
-                scale.unwrap_or(0), // and what else can we do here?
-            )),
+            Decimal(precision, scale) => {
```

Comment:
PyArrow supports decimal256 too, is there any reason this should error as opposed to outputting a decimal256 array?

Reply:
We don't support `Decimal256`. It turned out the precision given wasn't supported by Polars, so we raise earlier. Therefore we would also never need to `Decimal256`.